### PR TITLE
[rush] Add support for PNPM's --strict-peer-dependencies feature

### PR DESCRIPTION
**Pull request by pgonzal**

### CODE REVIEW

File: common/reviews/api/rush-lib.api.ts [hunk 1]
```diff
@@ -125,6 +125,14 @@ class LockStepVersionPolicy extends VersionPolicy {
   readonly version: string;
 }
 
+// @public
+class PnpmOptionsConfiguration {
+  // WARNING: The type "IPnpmOptionsJson" needs to be exported by the package (e.g. added to index.ts)
+  // @internal
```
Comment (nickpape-msft) [inline common/reviews/api/rush-lib.api.ts]: >internal [](start = 6, length = 8)

@private ?

**Comment by pgonzal** [comment]
@patmill FYI

File: common/reviews/api/rush-lib.api.ts (hunk 1 above)
Comment (pgonzal) [reply common/reviews/api/rush-lib.api.ts]: Good idea.  But this is an API Extractor feature request that's out of scope for this PR.

---
In reply to: [216065927](https://github.com/Microsoft/web-build-tools/pull/817#discussion_r216065927) [](ancestors = 216065927)

**Comment by pgonzal** [comment]
:shipit:

**Pull request merged**